### ISSUE TITLE
Make chunk caching non-optional

### ISSUE THREAD

**Issue by dktapps** [pmmp/PocketMine-MP#1448]
### Issue description
Since the changes made in https://github.com/pmmp/PocketMine-MP/commit/60f5c7ccefa2aba8e172902e5097eee61d2d37a7 , the memory footprint of the chunk cache is so small it's nonsensical to disable it.

Some background: "Chunk caching" refers to the retention of encoded FullChunkDataPackets so that they can be reused. This can save significant amounts of CPU time when lots of players join and/or are moving around.

Previously to the referenced commit, the cache stored uncompressed chunk packets, which meant that the memory usage would be doubled when the cache was enabled (which is why it was optional).
Since chunk packets have to be freshly prepared if a chunk is modified, this means the cache is _most_ useful in static worlds where blocks don't change very often and lots of players join in the same place. This meant that the chunk cache **used to be** a trade-off between higher memory usage and performance.

However, now, for performance reasons, the chunk prepare task now also pre-compresses the chunk before adding it to the cache, which does several things:
- It makes use of all available worker threads for compression, which improves main thread performance
- It significantly improves the performance benefits of the cache by not having to re-compress the chunk every time it is sent
- as well as reducing the memory footprint of a cached chunk packet to less than 2% of the size of an 
 in-memory chunk.

TL;DR: Chunk cache is now super lightweight and offers significant performance bonuses. It doesn't make any sense to allow disabling it anymore.
